### DISCUSSION TITLE
`open` doesn't return block result

### DISCUSSION THREAD

**Discussion by toy** [sparklemotion/sqlite3-ruby#414] [Ideas]
Normally I expect `open` method to initialise something, yield it, ensure closing and return result of block. For example for `File` I will get 10 bytes from `/dev/zero` in result variable:
```ruby
result = File.open('/dev/zero') do |f|
  f.read(10)
end
```
This is not the case for `SQLite3::Database.open`, as it is an alias to method new, so when called with block, it will return the already closed instance of `SQLite3::Database` instead of block result.
Is there a reason not to return block result? Otherwise what do you think about changing this? From one side it is a breaking change, as someone maybe relying on it, from the other side - it is not explicitly documented that open will not return block result when called with block.

**Comment by flavorjones** [comment]
@toy Thanks for asking this question. This behavior dates back to 2009 (faca48af), and I concede it's not documented (although it does have test coverage).

I think what you're suggesting would be an improvement. While I think there's a chance this change would break some applications, that likelihood feels small to me, and it would be easy to fix (change `open` to `new`). Generally I would support this change, though I'm the newest maintainer and folks like @tenderlove might want to weigh in.

Would you be interested in drafting a pull request with the new behavior?

**Reply by toy** [reply]
Thank you for quick reply, I'll draft a pull request

**Reply by toy** [reply]
The test was deleted in 2061408b2155b3517c4e9d79393f0484040645e3

**Reply by toy** [reply]
Drafted #415

**Reply by flavorjones** [reply]
Merged! Will be in the next release.
